pkgs/test_core/lib/src/runner/vm/platform.dart

```diff
@@ -63,8 +69,15 @@ class VMPlatform extends PlatformPlugin {
 
     VmService? client;
     StreamSubscription<Event>? eventSub;
-    var channel = IsolateChannel.connectReceive(receivePort)
-        .transformStream(StreamTransformer.fromHandlers(handleDone: (sink) {
+    var outerChannel = MultiChannel(IsolateChannel.connectReceive(receivePort));
```

Comment:
It would be good do document the flow here in comments since its a bit hard to follow with all the multi channels, and stream queues and things.

Reply:
I added a description of `channel` and `outerChannel` in a comment.

Longer term I think I can refactor a bit here, but I'll wait on that for now.